src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -174,7 +176,7 @@ public override int TestCaseCount
         /// <summary>
         /// The arguments to use in creating the fixture
         /// </summary>
-        public object[] Arguments { get; internal set; }
+        public override object[] Arguments { get { return arguments; } }
```

Comment:
Unfortunately with the change above, it is not possible to add setter to overridden property if abstract/virtual version of it does not have setter.

As the `TestMethod` implementation does not use setters, i.e. it is defined as follows:
```c#
public override object[] Arguments
{
    get { return parms != null ? parms.Arguments : null; }
}
```
I did not wanted to add setter to `Test` class as it will have no sense from `TestMethod` perspective.
With those constraints in mind I implemented `TestSuite` with separate `SetArguments()` method.

I'll take a second look at TestSuite to check if I can alter it so it will not expose setter for Arguments but instead accept them in constructor.